tools/c7n_azure/c7n_azure/functionapp_utils.py

```diff
@@ -125,102 +116,15 @@ def publish_functions_package(cls, function_params, package):
-                datetime.timedelta(days=FUNCTION_PACKAGE_SAS_EXPIRY_DAYS)
-                # expire in 10 years
-            )
-            blob_url = blob_client.make_blob_url(
-                FUNCTION_CONSUMPTION_BLOB_CONTAINER,
-                blob_name,
-                sas_token=sas)
-
-            # update application settings function package
-            app_settings = web_client.web_apps.list_application_settings(
-                function_params.function_app_resource_group_name,
-                function_params.function_app_name)
-            app_settings.properties['WEBSITE_RUN_FROM_PACKAGE'] = blob_url
-            web_client.web_apps.update_application_settings(
-                function_params.function_app_resource_group_name,
-                function_params.function_app_name,
-                kind=str,
-                properties=app_settings.properties
-            )
-
-            # Sync the scale controller for the Function App.
-            # Not required for the dedicated plans.
-            cls._sync_function_triggers(function_params)
-
-        cls.log.info('Finished publishing Function application')
+        publish_creds = web_client.web_apps.list_publishing_credentials(
+            function_params.function_app_resource_group_name,
+            function_params.function_app_name).result()
 
-    @classmethod
-    def _sync_function_triggers(cls, function_params):
-        cls.log.info('Sync Triggers...')
-        # This delay replicates behavior of Azure Functions Core tool
-        # Link to the github: https://bit.ly/2K5oXbS
-        time.sleep(5)
-        session = local_session(Session)
-        web_client = session.client('azure.mgmt.web.WebSiteManagementClient')
+        if package.wait_for_status(publish_creds):
+            package.publish(publish_creds)
 
-        max_retry_attempts = 3
-        for r in range(max_retry_attempts):
-            res = None
-            try:
-                res = web_client.web_apps.sync_function_triggers(
-                    function_params.function_app_resource_group_name,
-                    function_params.function_app_name
-                )
-            except (HttpOperationError, CloudError) as e:
-                # This appears to be a bug in the API
-                # Success can be either 200 or 204, which is
-                # unexpected and gets rethrown as a CloudError
-                if e.response.status_code in [200, 204]:
-                    return True
-
-                cls.log.error("Failed to sync triggers...")
-                cls.log.error(e)
-
-            if res and res.status_code in [200, 204]:
-                return True
-            else:
-                cls.log.info("Retrying in 5 seconds...")
-                time.sleep(5)
-
-        cls.log.error("Unable to sync triggers...")
-        return False
+            is_consumption = cls.is_consumption_plan(function_params)
+            package.wait_for_remote_build(publish_creds, is_consumption)
```

Comment:
so just wondering are we waiting at all for a reason if we're nto syncing triggers? i can see waiting in the tests, to verify, but on a deploy if we're not doing anything, can we just move on to the next?

Reply:
getting rid of the wait function here, would lift the coverage on this pr up to pass ci, its unclear thats its really adding anything per se.

Reply:
Afraid failure rate is high enough that it is important to wait and see if it actually deployed :)